openmetadata-service/src/main/java/org/openmetadata/service/util/DefaultTemplateProvider.java

```diff
@@ -0,0 +1,43 @@
+package org.openmetadata.service.util;
+
+import static freemarker.template.Configuration.VERSION_2_3_28;
+
+import freemarker.template.Configuration;
+import freemarker.template.Template;
+import java.io.IOException;
+import org.openmetadata.schema.email.SmtpSettings;
+import org.openmetadata.schema.settings.SettingsType;
+import org.openmetadata.service.resources.settings.SettingsCache;
+
+public class DefaultTemplateProvider implements TemplateProvider {
+  public static String EMAIL_TEMPLATE_BASEPATH = "/emailTemplates";
+  private static final Configuration templateConfiguration = new Configuration(VERSION_2_3_28);
+
+  public DefaultTemplateProvider() {
+    initializeTemplateConfiguration();
+  }
+
+  @Override
+  public void initializeTemplateConfiguration() {
+    String templatePath = getTemplatePath();
+
+    if (templatePath != null && !templatePath.isBlank()) {
+      EMAIL_TEMPLATE_BASEPATH += "/" + templatePath;
+    }
+
+    templateConfiguration.setClassForTemplateLoading(
+        DefaultTemplateProvider.class, EMAIL_TEMPLATE_BASEPATH);
+  }
+
+  @Override
+  public Template getTemplate(String templateName) throws IOException {
+    return templateConfiguration.getTemplate(templateName);
+  }
+
+  public static String getTemplatePath() {
+    SmtpSettings emailConfig =
+        SettingsCache.getSetting(SettingsType.EMAIL_CONFIGURATION, SmtpSettings.class);
```

Comment:
@harshach We are reading the configurations from the SystemSettings only, which are being set from the UI. In the front end, we need to set this field templatePath and send its value along with the other configuration values.

Reply:
I don't want email settings to be a server side config at all. Lets drive everything through the APIs and add a ./openmetadata-ops.sh that can pickup environment variables and update the email configs.

Reply:
Noted.